Makefile

```diff
@@ -50,59 +51,28 @@ ifndef HAS_PACKR2
 	go get -u github.com/gobuffalo/packr/v2/packr2
 endif
 
-xbuild-all:
+xbuild-all: xbuild-porter xbuild-mixins
+
+xbuild-porter: generate
 	$(MAKE) $(MAKE_OPTS) xbuild-all MIXIN=porter -f mixin.mk BINDIR=bin
-	$(MAKE) $(MAKE_OPTS) xbuild-all MIXIN=exec -f mixin.mk
-	$(MAKE) $(MAKE_OPTS) xbuild-all MIXIN=kubernetes -f mixin.mk
 
-xbuild-runtime:
-	$(MAKE) $(MAKE_OPTS) xbuild-runtime MIXIN=porter -f mixin.mk BINDIR=bin
-	$(MAKE) $(MAKE_OPTS) xbuild-runtime MIXIN=exec -f mixin.mk
-	$(MAKE) $(MAKE_OPTS) xbuild-runtime MIXIN=kubernetes -f mixin.mk
+xbuild-mixins: $(addprefix xbuild-mixin-,$(INT_MIXINS))
+xbuild-mixin-%: generate
+	$(MAKE) $(MAKE_OPTS) xbuild-all MIXIN=$* -f mixin.mk
 
-xbuild-client:
-	$(MAKE) $(MAKE_OPTS) xbuild-client MIXIN=porter -f mixin.mk BINDIR=bin
-	$(MAKE) $(MAKE_OPTS) xbuild-client MIXIN=exec -f mixin.mk
-	$(MAKE) $(MAKE_OPTS) xbuild-client MIXIN=kubernetes -f mixin.mk
+get-mixins: $(addprefix get-mixin-,$(EXT_MIXINS))
```

Comment:
You might've already played around with this option, but we could reduce this and the `get-mixin-*` targets below with:
```make
get-mixins:
	$(foreach MIXIN, $(EXT_MIXINS), \
		bin/porter mixin install $(MIXIN) --version $(MIXIN_TAG) --url $(MIXINS_URL)/$(MIXIN); \
	)
```
Which would be handy as the external mixin list grows (just need to update `EXT_MIXINS`, etc.)

Though, if there is a common use case for needing to fetch individual external mixins, this might not be the best approach.

Reply:
I did it this way originally because I thought it would help for getting an individual mixin. But really the command being run `porter mixin install` isn't hard. So if someone needed to be doing one off work on a single mixin, then I don't see why they couldn't use that directly instead of relying on the makefile.

So yeah let me make that change.